urle381b6cba7_IkarosKappler_ikrs_http_tgz-pJ8-ikrs_json_rpc_JSONRPCHandlerJ8/src/ikrs/yuccasrv/socketmngr/ServerSocketThread.java

```diff
@@ -0,0 +1,468 @@
+ * basic server into a concurrent thread.
+ *
+ *
+ * @author Ikaros Kappler
+ * @date 2012-04-22
+ * @version 1.0.0
+ **/
+
+
+public class ServerSocketThread
+    extends Thread
+    implements Runnable {
+
+    public static final int DEFAULT_BACKLOG = 10;
+
+    private InetAddress bindAddress;
+    private int bindPort;
+    private Environment<String,BasicType> bindSettings;
+
+    private ServerSocket serverSocket;
+
+    private UUID uuid;
+
+    private ServerSocketThreadObserver observer;
+
+    private CustomLogger logger;
+
+    /**
+     * The constructor.
+     **/
+    protected ServerSocketThread( CustomLogger logger,
+				  InetAddress bindAddress,
+				  int bindPort,
+				  Environment<String,BasicType> bindSettings, 
+				  ServerSocketThreadObserver observer 				  
+				  ) 
+	throws IOException,
+	       GeneralSecurityException {	
+
+	super();
+
+	this.logger = logger;
+
+	this.bindAddress = bindAddress;
+	this.bindPort = bindPort;
+	this.bindSettings = bindSettings;
+	
+	this.observer = observer;
+
+	int backlog = DEFAULT_BACKLOG;
+	if( bindSettings.get(Constants.CONFIG_SERVER_BACKLOG) != null )
+	    backlog = bindSettings.get(Constants.CONFIG_SERVER_BACKLOG).getInt( DEFAULT_BACKLOG );
+
+	boolean useSSL = false;
+	if( bindSettings.get(Constants.CONFIG_SERVER_SSL) != null )
+	    useSSL = bindSettings.get(Constants.CONFIG_SERVER_SSL).getBoolean();
+
+
+
+	if( useSSL ) {
+
+	    try {
+		this.serverSocket = createSSLServerSocket( bindPort,
+							   backlog,
+							   bindAddress,
+							   this.bindSettings 
+							   );
+	    } catch( NoSuchAlgorithmException e ) {
+
+		throw new GeneralSecurityException( "Internal error: Cannot create SSLServerSocket. Failed to load encryption algorithm.", e );
+
+	    }
+
+	} else {
+	    
+	    this.serverSocket = new ServerSocket( bindPort,
+						  backlog,
+						  bindAddress 
+						  );
+	}
+
+	this.uuid = UUID.randomUUID();
+
+    }
+
+    private SSLServerSocket createSSLServerSocket( int bindPort,
+						   int backlog,
+						   InetAddress bindAddress,
+						   Environment<String,BasicType> bindSettings ) 
+	throws IOException,
+	       FileNotFoundException,
+	       KeyStoreException,
+	       KeyManagementException,
+	       NoSuchAlgorithmException,
+	       CertificateException,
+	       UnrecoverableKeyException {
+
+
+	BasicType wrp_keyStoreName    = getSSLConfigPropertyValue( bindSettings, Constants.CONFIG_SERVER_SSL_KEYSTORE );
+	BasicType wrp_keyStorePass    = getSSLConfigPropertyValue( bindSettings, Constants.CONFIG_SERVER_SSL_KEYSTOREPASSWORD );
+	BasicType wrp_keyStoreType    = getSSLConfigPropertyValue( bindSettings, Constants.CONFIG_SERVER_SSL_KEYSTORETYPE );
+
+	BasicType wrp_trustStoreName  = getSSLConfigPropertyValue( bindSettings, Constants.CONFIG_SERVER_SSL_TRUSTSTORE );
+	BasicType wrp_trustStorePass  = getSSLConfigPropertyValue( bindSettings, Constants.CONFIG_SERVER_SSL_TRUSTSTOREPASSWORD );
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears your code contains a hardcoded password. Hardcoded secrets or credentials can allow attackers to bypass authentication methods and perform malicious actions. We recommend revoking access to resources using this credential and storing future credentials in a management service such as [AWS Secrets Manager](https://aws.amazon.com/secrets-manager/).

[Learn more about the use of hardcoded credentials](https://cwe.mitre.org/data/definitions/798.html)